src/robot/running/arguments/argumentparser.py

```diff
@@ -198,13 +199,15 @@ def parse(self, argspec, name=None):
                     self._raise_invalid_spec('Cannot have multiple varargs.')
                 self._add_varargs(spec, arg)
                 kw_only_args = True
-            elif '=' in arg:
+            elif self._have_default_value(arg):
                 self._add_arg_with_default(spec, arg, kw_only_args)
             elif spec.defaults and not kw_only_args:
                 self._raise_invalid_spec('Non-default argument after default '
                                          'arguments.')
             else:
                 self._add_arg(spec, arg, kw_only_args)
+            if type_:
+                self._add_type_to_arg(spec, arg, type_)
```

Comment:
Is it necessary to get the type information in the beginning? Wouldn't it be easier to just enhance `_add_arg` so that it can check does the name have embedded type information and then act accordingly?

Reply:
yes it is necessary because _split_args_and_types remove type in args. If we don't split in begin we have problem with kwargs and vargs when the argname will be 'arg:type'